### PR TITLE
Change `Series.divide/2` return float series and add `quotient/2` and `remainder/2` to `Series`

### PR DESCRIPTION
**Pull request by philss**
This is because we want to make it similar to the behaviour of Elixir
and Nx.

Related to #333

### CODE REVIEW

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -1254,20 +1254,24 @@ defmodule Explorer.Series do
...
-        [5, 5, 5]
+        float[3]
+        [5.0, 5.0, 5.0]
       >
   """
   @doc type: :element_wise
   @spec divide(left :: Series.t(), right :: Series.t() | number()) :: Series.t()
-  def divide(left, right), do: basic_numeric_operation(:divide, left, right)
+  def divide(%Series{} = left, right) do
+    left = with %Series{dtype: :integer} <- left, do: cast(left, :float)
```
Comment (josevalim) [inline lib/explorer/series.ex:1271]: We can be smarter and only cast if right is an integer series and an integer itself.

**Comment by philss** [comment]
I'm still working on this. I'm going to add `quotient` and `reminder` in this PR.

**Comment by cigrainger** [comment]
Looking good!

**Review by josevalim — APPROVED** [review]

**Review by josevalim — APPROVED** [review]

**Comment by philss** [comment]
@cigrainger please take a look again. I added the two new functions :)

File: lib/explorer/series.ex [hunk 2]
```diff
@@ -1250,24 +1250,137 @@ defmodule Explorer.Series do
...
-  def divide(left, right), do: basic_numeric_operation(:divide, left, right)
+  @spec quotient(left :: Series.t(), right :: Series.t() | integer()) :: Series.t()
+  def quotient(%Series{dtype: :integer} = left, %Series{dtype: :integer} = right),
+    do: Shared.apply_impl(left, :quotient, [right])
+
+  def quotient(%Series{dtype: :integer} = left, right) when is_integer(right),
+    do: Shared.apply_impl(left, :quotient, [right])
+
+  @doc """
+  Computes de remainder of an element-wise integer division.
```
Comment (cigrainger) [inline lib/explorer/series.ex:1343]:
```suggestion
  Computes the remainder of an element-wise integer division.
```

**Review by cigrainger — APPROVED** [review]
Just one thing in the docs, otherwise :shipit:

**Comment by josevalim** [comment]
@philss don't merge this yet, let's do a new v0.3.1 release with some fixes and then we merge it.

**Pull request merged**